js/ui/viewSelector.js

```diff
@@ -146,6 +151,33 @@ const ViewsDisplayLayout = new Lang.Class({
             searchResultsBox.y2 = searchResultsBox.y1 + searchResultsHeight;
             this._searchResultsActor.allocate(searchResultsBox, flags);
         }
+    },
+
+    _updateCompositeMode: function() {
+        /* Given that the height of the widget varies according to the
+         * number of icons, we only check the width here.
+         */
+        this.compositeMode = Main.layoutManager.primaryMonitor &&
+                             Main.layoutManager.primaryMonitor.height < 600;
+    },
+
+    set compositeMode(mode) {
+        if (this._compositeMode != mode) {
+            this._compositeMode = mode;
+
+            /* When running on small screens, to make it fit 3 rows of icons,
+             * reduce the space above and below the search entry by adding (or
+             * removing, in case the screen is big enough) the .composite-mode
```

Comment:
Remember to update the comment if you move to use something other than "composite-mode" / "compositeMode".